sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/checkin/SentryCheckInAdvice.java

```diff
@@ -86,7 +87,7 @@ public Object invoke(final @NotNull MethodInvocation invocation) throws Throwabl
       return invocation.proceed();
     }
 
-    scopes.pushScope();
+    final @NotNull ISentryLifecycleToken lifecycleToken = scopes.pushIsolationScope();
```

Comment:
Not 100% sure here, maybe we should make it configurable whether this isolates or not (`pushScope` vs. `pushIsolationScope`).

Reply:
Default should probably be isolated.

Reply:
Yeah makes sense! 
If any user code within `invocation.proceed();` performs changes via the static API, e.g. `Sentry.setTag()`, those will be written to the isolation scope which is created here, right?

Reply:
Yeah, static API and `configureScope` default to isolation scope for non Android.